fe1-web/src/features/wallet/screens/WalletHome.tsx

```diff
@@ -1,51 +1,111 @@
-import { useNavigation } from '@react-navigation/native';
-import React from 'react';
+import PropTypes from 'prop-types';
+import React, { useEffect, useState } from 'react';
 import { StyleSheet, View, ViewStyle } from 'react-native';
+import { useSelector } from 'react-redux';
 
-import { TextBlock, WideButtonView } from 'core/components';
+import { QRCode, TextBlock, WideButtonView } from 'core/components';
 import containerStyles from 'core/styles/stylesheets/containerStyles';
+import { LaoEventType } from 'features/events/objects';
+import { makeEventByTypeSelector } from 'features/events/reducer';
+import { RollCall } from 'features/rollCall/objects';
+import PROPS_TYPE from 'resources/Props';
 import STRINGS from 'resources/strings';
 
-import { WalletStore } from '../store';
+import { RollCallTokensDropDown } from '../components';
+import * as Wallet from '../objects';
+import { useMockWalletState } from '../objects/__mocks__/mockWallet';
+import { RollCallToken } from '../objects/RollCallToken';
 
 const styles = StyleSheet.create({
+  homeContainer: {
+    ...containerStyles.centeredXY,
+    padding: '30px',
+  } as ViewStyle,
   smallPadding: {
     padding: '1rem',
   } as ViewStyle,
-  largePadding: {
-    padding: '2rem',
+  tokenSelectContainer: {
+    flexDirection: 'row',
+    marginTop: 30,
+    marginBottom: 20,
   } as ViewStyle,
 });
 
+const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
+
 /**
- * Wallet home screen
+ * Wallet UI once the wallet is synced
  */
-const WalletHome = () => {
-  // FIXME: Navigation should use a defined type here (instead of any)
-  const navigation = useNavigation<any>();
+const WalletHome = ({ navigation }: IPropTypes) => {
+  const [tokens, setTokens] = useState<RollCallToken[]>();
+  const [selectedToken, setSelectedToken] = useState<RollCallToken>();
+  const [isDebug, setIsDebug] = useState(false);
+
+  const rollCalls = useSelector(rollCallSelector);
+  const mock = useMockWalletState();
 
-  function importSeed() {
-    if (WalletStore.hasSeed()) {
-      navigation.navigate(STRINGS.navigation_synced_wallet);
+  useEffect(() => {
+    Wallet.recoverWalletRollCallTokens()
+      .then((rct) => {
+        setTokens(rct);
+        setSelectedToken(rct[0]);
+      })
+      .catch((e) => {
+        console.debug(e);
+      });
+  }, [rollCalls, isDebug]);
+
+  const setDebug = (isOn: boolean) => {
```

Comment:
Suggestion: Rename to `toggleDebugMode` and directly access the `isDebug` variable without accepting it as an argument?

I was a bit confused by the similar names at first 😅